### ISSUE TITLE
TotalF1 clarification for average=Macro use case

### ISSUE THREAD

**Issue by martinerrazquin** [catboost/catboost#2108]
catboost version: 1.0.6
CPU: X (for what it's worth)
GPU: -

## Problem
Hi! This is just a default value problem. When using the `class_weights` parameter for e.g. `CatBoostClassifier` in a MultiClassification context the TotalF1 metric with average=Macro will still use the class weights, defeating the purpose of macro-averaging.
From the [Macro-averaged F1 definition](https://catboost.ai/en/docs/concepts/loss-functions-multiclassification#macro) one would expect for it to not use weights (specially given the existence of the Weighted option); though from my understanding it still calculates each F1 value in a weighted manner, because that is the default value.
One can check with Scikit-Learn `classification_report` that the `TotalF1:average=Macro` metric matches the weighted-average F1 metric, while `TotalF1:average=Macro;use_weights=False` will correctly calculate the macro-average F1.  Although there are workarounds, I think this case is common enough to merit a clarification of some sort.

## TL;DR

When using e.g. `CatBoostClassifier` with `class_weight` parameter (at least in a MultiClassification context) using the `TotalF1:average=Macro` metric will return the Weighted Total F1 instead of the Macro-averaged Total F1.
The workaround is using `TotalF1:average=Macro;use_weights=False` but this is what people look for when using the Macro-average, and should (in my opinion) therefore be either documented or have this specific default value changed.
